src/UrlMatcher.php

```diff
@@ -60,6 +60,16 @@ final class UrlMatcher implements UrlMatcherInterface
     private ?Route $currentRoute = null;
     private bool $hasInjectedRoutes = false;
 
+    /**
+     * Yii Framework encodes URLs differently than previous versions. If you are
+     * migrating a project from older versions, you can set this value to `true`
+     * to keep URLs encoded the same way.
+     * Default `false` is RFC 3986 compliant
+     *
+     * @var bool
+     */
+    private bool $yii2Compat = false;
```

Comment:
I don't think we need it.

Reply:
Then we need at least to name it differently.

Reply:
This function is not for those people. Also, in params there is a clear explanation when to set it to true.

Reply:
Well, I think we can name it better by what it does i.e. `encodeWithRFC` or `useUrlencode`, or `useRawurlencode` or something.

Reply:
What it does is making router package compatible with how Yii 2 deals with parameters. Below that it indeed chooses between `urlencode` and `rawurlencode`, but in my opinion those names you suggested does not make anything clearer.

I you pick another name I will of course follow your lead, I just wanted to make you aware of my reasoning behind the name.